// deep-common-lib/fesm5/deep-common-lib.js
import { Injectable, defineInjectable, Component, NgModule } from '@angular/core';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var DeepCommonLibService = /** @class */ (function () {
    function DeepCommonLibService() {
    }
    DeepCommonLibService.decorators = [
        { type: Injectable, args: [{
                    providedIn: 'root'
                },] }
    ];
    /** @nocollapse */
    DeepCommonLibService.ctorParameters = function () { return []; };
    /** @nocollapse */ DeepCommonLibService.ngInjectableDef = defineInjectable({ factory: function DeepCommonLibService_Factory() { return new DeepCommonLibService(); }, token: DeepCommonLibService, providedIn: "root" });
    return DeepCommonLibService;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var DeepCommonLibComponent = /** @class */ (function () {
    function DeepCommonLibComponent() {
    }
    /**
     * @return {?}
     */
    DeepCommonLibComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
    };
    DeepCommonLibComponent.decorators = [
        { type: Component, args: [{
                    selector: 'san-deep-common-lib',
                    template: "\n    <p>\n      deep-common-lib works!\n    </p>\n  "
                }] }
    ];
    /** @nocollapse */
    DeepCommonLibComponent.ctorParameters = function () { return []; };
    return DeepCommonLibComponent;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var DeepCommonLibModule = /** @class */ (function () {
    function DeepCommonLibModule() {
    }
    DeepCommonLibModule.decorators = [
        { type: NgModule, args: [{
                    declarations: [DeepCommonLibComponent],
                    imports: [],
                    exports: [DeepCommonLibComponent]
                },] }
    ];
    return DeepCommonLibModule;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

export { DeepCommonLibComponent, DeepCommonLibModule, DeepCommonLibService };
