// deep-common-lib/bundles/deep-common-lib.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core')) :
    typeof define === 'function' && define.amd ? define('deep-common-lib', ['exports', '@angular/core'], factory) :
    (global = global || self, factory(global['deep-common-lib'] = {}, global.ng.core));
}(this, function (exports, core) { 'use strict';

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var DeepCommonLibService = /** @class */ (function () {
        function DeepCommonLibService() {
        }
        DeepCommonLibService.decorators = [
            { type: core.Injectable, args: [{
                        providedIn: 'root'
                    },] }
        ];
        /** @nocollapse */
        DeepCommonLibService.ctorParameters = function () { return []; };
        /** @nocollapse */ DeepCommonLibService.ngInjectableDef = core.defineInjectable({ factory: function DeepCommonLibService_Factory() { return new DeepCommonLibService(); }, token: DeepCommonLibService, providedIn: "root" });
        return DeepCommonLibService;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var DeepCommonLibComponent = /** @class */ (function () {
        function DeepCommonLibComponent() {
        }
        /**
         * @return {?}
         */
        DeepCommonLibComponent.prototype.ngOnInit = /**
         * @return {?}
         */
        function () {
        };
        DeepCommonLibComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'san-deep-common-lib',
                        template: "\n    <p>\n      deep-common-lib works!\n    </p>\n  "
                    }] }
        ];
        /** @nocollapse */
        DeepCommonLibComponent.ctorParameters = function () { return []; };
        return DeepCommonLibComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var DeepCommonLibModule = /** @class */ (function () {
        function DeepCommonLibModule() {
        }
        DeepCommonLibModule.decorators = [
            { type: core.NgModule, args: [{
                        declarations: [DeepCommonLibComponent],
                        imports: [],
                        exports: [DeepCommonLibComponent]
                    },] }
        ];
        return DeepCommonLibModule;
    }());

    exports.DeepCommonLibComponent = DeepCommonLibComponent;
    exports.DeepCommonLibModule = DeepCommonLibModule;
    exports.DeepCommonLibService = DeepCommonLibService;

    Object.defineProperty(exports, '__esModule', { value: true });

}));
