// deep-common-lib/fesm2015/deep-common-lib.js
import { Injectable, defineInjectable, Component, NgModule } from '@angular/core';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class DeepCommonLibService {
    constructor() { }
}
DeepCommonLibService.decorators = [
    { type: Injectable, args: [{
                providedIn: 'root'
            },] }
];
/** @nocollapse */
DeepCommonLibService.ctorParameters = () => [];
/** @nocollapse */ DeepCommonLibService.ngInjectableDef = defineInjectable({ factory: function DeepCommonLibService_Factory() { return new DeepCommonLibService(); }, token: DeepCommonLibService, providedIn: "root" });

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class DeepCommonLibComponent {
    constructor() { }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
DeepCommonLibComponent.decorators = [
    { type: Component, args: [{
                selector: 'san-deep-common-lib',
                template: `
    <p>
      deep-common-lib works!
    </p>
  `
            }] }
];
/** @nocollapse */
DeepCommonLibComponent.ctorParameters = () => [];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class DeepCommonLibModule {
}
DeepCommonLibModule.decorators = [
    { type: NgModule, args: [{
                declarations: [DeepCommonLibComponent],
                imports: [],
                exports: [DeepCommonLibComponent]
            },] }
];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

export { DeepCommonLibComponent, DeepCommonLibModule, DeepCommonLibService };
